controller/controller.go

```diff
@@ -163,7 +163,21 @@ func (c *Controller) DeleteFunc(obj interface{}) {
 	resourceSet, err := c.resourceRouter.ResourceSet(obj)
 	if IsNoResourceSet(err) {
 		// In case the resource router is not able to find any resource set to
-		// handle the reconciled runtime object, we stop here.
+		// handle the reconciled runtime object, we stop here. Note that we just
+		// remove the finalizer regardless because at this point there will never be
+		// a chance to remove it otherwhise because nobody wanted to handle this
+		// runtime object anyway.
+
+		c.logger.Log("level", "debug", "message", "removing finalizer from runtime object")
+
+		err = c.removeFinalizer(obj)
+		if err != nil {
+			c.logger.Log("level", "error", "message", "stop reconciliation due to error", "stack", fmt.Sprintf("%#v", err))
+			return
+		}
+
+		c.logger.Log("level", "debug", "message", "removed finalizer from runtime object")
```

Comment:
I am just adding this here again. Alternative would be to have some conditional magic below but I thought this is more straight forward wit the "early handle".